### PR TITLE
[SEDONA-399] Add H3 functions

### PR DESCRIPTION
**Pull request by zongsizhang**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-399

## What changes were proposed in this PR?
Integrate Uber H3 into Sedona Common, Sql and Flink
Support turning geometries into Uber H3 cells. 
Wrapped the basic cells processes functions into Sedona Sql Functions.

In short, the following functions are added:
* ST_H3CellDistance
* ST_H3CellIDs
* ST_H3ToGeom
* ST_H3KRing

## How was this patch tested?
Test cases with geomery datasets

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation update.

### CODE REVIEW

**Comment by jiayuasu** [comment]
This will close Issue https://github.com/apache/sedona/issues/914

**Comment by MoSarwat** [comment]
Are H3 docs included in the PR

**Comment by jiayuasu** [comment]
@MoSarwat Yes it is included
